src/Sylius/Bundle/PayumBundle/Validator/GatewayFactoryExistsValidator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PayumBundle\Validator;
+
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Sylius\Bundle\PayumBundle\Validator\Constraints\GatewayFactoryExists;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class GatewayFactoryExistsValidator extends ConstraintValidator
+{
+    public const OFFLINE_GATEWAY_FACTORY = 'offline';
+
+    /** @param array<string, string> $factoryNames */
+    public function __construct(private array $factoryNames)
+    {
+    }
+
+    /** @param GatewayConfigInterface $value */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$value instanceof GatewayConfigInterface) {
+            throw new UnexpectedValueException($value, GatewayConfigInterface::class);
+        }
+
+        if (!$constraint instanceof GatewayFactoryExists) {
+            throw new UnexpectedTypeException($constraint, GatewayFactoryExists::class);
+        }
+
+        /** @var string|null $factoryName */
+        $factoryName = $value->getFactoryName();
+
+        if ($factoryName === self::OFFLINE_GATEWAY_FACTORY || $factoryName === null) {
```

Comment:
An empty string would skip this yielding 2 errors since there's also a `NotBlank` on the property itself